src/program.hpp

```diff
@@ -219,12 +231,33 @@ class spir_binary {
         m_constant_data_buffer.reset(new constant_data_buffer_info(info));
     }
 
+    void add_image_channel_order_metadata(const std::string& name,
+                                          uint32_t ordinal, uint32_t offset) {
+        if (m_image_channel_metadata[name].count(ordinal) > 0) {
+            m_image_channel_metadata[name][ordinal].first = offset;
+        } else {
+            m_image_channel_metadata[name][ordinal] =
+                std::make_pair(offset, IMAGE_METADATA_EMPTY);
```

Comment:
If the metadata were a struct you could rely on the default constructor and all that special handling would not be necessary.

Reply:
Done